import/index_java/src/org/vufind/index/FormatCalculator.java

```diff
@@ -466,29 +608,35 @@ protected List<String> getFormatsAsList(Record record) {
         }
 
         // check the Leader at position 6
-        char recordType = Character.toLowerCase(leader.charAt(6));
         if (definitelyNotBookBasedOnRecordType(recordType, marc008)) {
             couldBeBook = false;
         }
-        String formatFromRecordType = getFormatFromRecordType(record, recordType);
+        String formatFromRecordType = getFormatFromRecordType(record, recordType, marc008, formatCodes007);
         if (formatFromRecordType.length() > 0) {
             result.add(formatFromRecordType);
         }
 
         // check the Leader at position 7
-        char bibLevel = Character.toLowerCase(leader.charAt(7));
         String formatFromBibLevel = getFormatFromBibLevel(
-            record, bibLevel, formatCode, marc008, couldBeBook
+            record, recordType, bibLevel, marc008, couldBeBook, formatCodes007
```

Comment:
Is there a strong reason that you changed the position of the bibLevel parameter? If you did this for logical reasons (e.g. recordType is broader and bibLevel is more specific) then I have no objections to it, but if it just happened by chance, it might be worth swapping the order back to reduce the amount of change to the signature, to make adjustment of local customizations slightly less complicated.

Reply:
LDR/06 < LDR/07

So a bit esoteric. Happy to move it to the end for reasons you say.

Reply:
I'll leave it to your discretion. I don't mind leaving it as-is for the reason you cite, especially since the signature is changing for other reasons regardless.

Reply:
No objections here -- thanks for talking it through!